packages/bee-q/src/components/dropdown/bq-dropdown.tsx

```diff
@@ -0,0 +1,160 @@
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** Represents the distance (gutter or margin) between the panel and the trigger element. */
+  @Prop({ reflect: true }) distance?: number = 4;
+
+  /** Position of the panel */
+  @Prop({ reflect: true }) placement?: FloatingUIPlacement = 'bottom-start';
+
+  /** If true, the panel will be visible. */
+  @Prop({ reflect: true }) open?: boolean = false;
+
+  /** The trigger element for the panel */
+  @Prop() triggerElement?: HTMLElement;
+
+  /** Whether the panel should have the same width as the trigger element */
+  @Prop({ reflect: true }) sameWidth?: boolean = false;
+
+  /**  Represents the skidding between the panel and the trigger element. */
+  @Prop({ reflect: true }) skidding?: number = 0;
+
+  /** Defines the strategy to position the panel */
+  @Prop({ reflect: true }) strategy?: 'fixed' | 'absolute' = 'fixed';
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /**
+   * Handler to be called to check if the `bq-panel` switches state (visible/hidden).
+   * @returns CustomEvent - with value `{ opened: boolean }`
+   */
+  @Event() bqPanelChange: EventEmitter<{ opened: boolean }>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  /** Listens for the 'click' event on the document object
+   * and closes the dropdown panel if the click is outside the component.
+   */
+  @Listen('click', { target: 'document', passive: true })
+  onClickOutside(event: MouseEvent) {
+    if (!this.open) return;
+
+    // Close when clicking outside of the close element
+    const path = event.composedPath();
+    if (!path.includes(this.el)) {
+      this.open = false;
+    }
+  }
+
+  /**
+   * Listens for the 'keyup' event on the window object
+   * and closes the dropdown panel if the 'Escape' key or 'Tab' key outside the component is pressed.
+   */
+  @Listen('keyup', { target: 'window', passive: true })
+  onEscape(event: KeyboardEvent) {
+    if (!this.open) return;
+
+    if (event.key === 'Escape' || (event.key === 'Tab' && !event.composedPath().includes(this.el))) {
+      this.open = false;
+    }
+  }
+
+  @Listen('bqSelect', { passive: true })
+  onItemSelect() {
+    this.open = false;
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private togglePanel = (): void => {
+    this.open = !this.open;
+  };
```

Comment:
Another good improvement will be adding a property that allows users to set whether the panel dropdown should stay open or not after selection. E.g:

```js
  /** If true, the panel will remain open after a selection is made. */
  @Prop({ reflect: true }) keepOpenOnSelect?: boolean = false;

  ...

  @Listen('bqSelect', { passive: true })
  onItemSelect() {
    if (this.keepOpenOnSelect) return;

    this.open = false;
  }
```

Reply:
Like this!